ProcessResultsUtils.scala

```diff
@@ -0,0 +1,147 @@
+
+
+  def addTableToSpark (spark: SparkSession, config: Configuration,
+                       dataConfigName: String) : Unit = {
+    LOGGER.info("addTableToSpark")
+    var tableDF = loadTableDFFromConfig(null, spark, config, dataConfigName)
+    var sparkName = dataConfigName.toString()
+    if (dataConfigName.toString().contains("clauses")) {
+      sparkName = "clauses"
+    }
+    tableDF.createOrReplaceTempView(sparkName)
+    if (debugMode) {
+      println("============= " + sparkName + " ===============")
+      tableDF.show(false)
+    }
+  }
+
+  def createJoinDataFrame (spark: SparkSession, query:String, config: Configuration, testType: String): Unit = {
+
+    val json: JsValue = Json.parse(query)
+    val table: String = new String("table")
+    var index: Integer = 0;
+    var cond = true;
+    while (cond) {
+      var tableKey = table + index.toString
+      index = index + 1
+      val tableConfigName = (json \ tableKey).validate[String]
+      tableConfigName match {
+        case s: JsSuccess[String] => addTableToSpark(spark, config, s.get);
+        case e: JsError => cond = false
+      }
+    }
+    val newQuery = (json \ "newQuery").validate[String]
+    if (debugMode) {
+      println("the re-written query: " + newQuery.get)
+    }
+    val bloodTestsDF = spark.sql(newQuery.get).toDF().filter(row => anyNotNull(row))
+    if (debugMode) {
+      bloodTestsDF.show(false)
+    }
+    val profilesDF = loadTableDFFromConfig(null, spark, config, "patientsProfiles")
+    if (debugMode) {
+      profilesDF.show(false)
+    }
+    //TODO: check if inner join can be applied
+    var joinedDF = bloodTestsDF.join(profilesDF, bloodTestsDF.col(Constants.SUBJECT_ID_COL_NAME).
+      equalTo(profilesDF.col(Constants.SUBJECT_ID_COL_NAME)), "left_outer")
+    joinedDF = joinedDF.drop(profilesDF.col(Constants.SUBJECT_ID_COL_NAME))
+    joinedDF.createOrReplaceTempView("joined")
+    if (debugMode) {
+      joinedDF.show(false)
+    }
+  }
+
+  def getBloodTestsComponentCompilantResult (spark: SparkSession, query:String, config: Configuration, testType: String,
+                                             patientSSN: String): String = {
+
+    createJoinDataFrame(spark, query, config, testType)
+    val queryOnJoinTables = "SELECT patientId, date, %s FROM joined WHERE socialId=\"%s\"".format(testType, patientSSN)
+    var patientBloodTestsDF = spark.sql(queryOnJoinTables).toDF().filter(row => anyNotNull(row))
+    if (debugMode) {
+      patientBloodTestsDF.limit(10).show(false)
+      patientBloodTestsDF.printSchema
+      patientBloodTestsDF.explain(true)
+    }
+    //Adjust output to blueprint
+    patientBloodTestsDF = patientBloodTestsDF.withColumnRenamed(testType, "value").drop(Constants.SUBJECT_ID_COL_NAME).distinct()
+    patientBloodTestsDF.toJSON.collect.mkString("[", ",", "]")
+  }
+
+  def getAvgBloodTestsTestTypeCompilantResult (spark: SparkSession, query:String, config: Configuration, testType: String, avgTestType: String,
```

Comment:
No need in all these methods getBloodTestsComponentCompilantResult , getAvgBloodTestsTestTypeCompilantResult . They just differ in query string.
Business logic should not be located inside the utilites class...